index.js

```diff
@@ -79,8 +152,52 @@ function sparqlProxy (options) {
       }, options.timeout)
     }
 
-    return client[queryOperation](query, currentQueryOptions).then((result) => {
+    if (cacheEnabled) {
+      cacheKey = cacheKeyName(cachePrefix, sha1(query))
+      const cacheResponse = await cacheClient.get(cacheKey)
+      const jsonResponse = JSON.parse(cacheResponse)
+      if (cacheResponse) {
+        const cacheData = jsonResponse.data || ''
```

Comment:
Then if the function returns for real calls and cache calls the same, let's say `jsonResponse` the following code can be the same for both.

Reply:
Another approach is to have two functions, one of them with cache that might return something or not, and the 'real one'

const result = tryRequestWithCache(query) ?? tryRequest(query)